From: SeongJae Park <sj@kernel.org>
Date: Mon, 20 Jan 2025 13:43:21 -0800
Subject: [PATCH] ==== damon_call() for online parameter commit ====

mm/damon/sysfs: commit parameters online via damon_call()

Due to the lack of ways to synchronously access DAMON internal data,
DAMON sysfs interface is using damon_callback hooks with its own
synchronization mechanism.  The mechanism is built on top of
damon_callback hooks in an ineifficient and complicated way.

Patch series "mm/damon: replace most damon_callback usages in sysfs with
new core functions", which starts with commit e035320fd38e
("mm/damon/sysfs-schemes: remove unnecessary schemes existence check in
damon_sysfs_schemes_clear_regions()") introduced two new DAMON kernel
API functions that providing the synchronous access, replaced most
damon_callback hooks usage in DAMON sysfs interface, and cleaned up
unnecessary code.

Continue the replacement and cleanup works.  Update the last DAMON
sysfs' usage of its own synchronization mechanism, namely online DAMON
parameters commit, to use damon_call() instead of the damon_callback
hooks and the hard-to-maintain core-external synchronization mechanism.
Then remove the no more be used code due to the change, and more unused
code that just not yet cleaned up.

The first four patches (patches 1-4) of this series makes DAMON sysfs
interface's online parameters commit to use damon_call().  Then,
following three patches (patches 5-7) remove the DAMON sysfs interface's
own synchronization mechanism and its usages, which is no more be used
by anyone due to the first four patches.  Finally, six patches (8-13) do
more cleanup of outdated comment and unused code.

Changes from RFC
(https://lore.kernel.org/20250226063651.513178-1-sj@kernel.org)
- Fix aggregated information double reset problem
- Wordsmith commit messages
- Rebase to latest mm-unstable

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_cw2l52w8 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_cw2l52w8

diff --git a/hkml_cv_bogus/hkml_cv_bogus_cw2l52w8 b/hkml_cv_bogus/hkml_cv_bogus_cw2l52w8
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

